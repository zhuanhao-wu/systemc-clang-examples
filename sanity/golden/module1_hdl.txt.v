module topadd2_0(
input wire [31:0] in_port_1,
input wire [31:0] in_port_2,
input wire clk,
output reg [31:0] out_port 
);


always @(posedge clk) begin: topEntry

out_port <= (in_port_1) + (in_port_2);
end // topEntry
endmodule // topadd2_0

