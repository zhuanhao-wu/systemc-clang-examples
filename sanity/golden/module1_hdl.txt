  hModule topadd2_0 [
    hPortsigvarlist  NONAME [
      hPortin in_port_1 [
        hTypeinfo  NONAME [
          hType "sc_in" NOLIST
          hType "int" NOLIST
        ]
      ]
      hPortin in_port_2 [
        hTypeinfo  NONAME [
          hType "sc_in" NOLIST
          hType "int" NOLIST
        ]
      ]
      hPortin clk [
        hTypeinfo  NONAME [
          hType "sc_in" NOLIST
          hType "_Bool" NOLIST
        ]
      ]
      hPortout out_port [
        hTypeinfo  NONAME [
          hType "sc_out" NOLIST
          hType "int" NOLIST
        ]
      ]
    ]
    hProcesses  NONAME [
      hProcess topEntry [
        hSenslist  NONAME [
          hSensvar clk [
            hSensedge pos NOLIST
          ]
        ]
        hCStmt  NONAME [
          hSigAssignL write [
            hLiteral out_port NOLIST
            hBinop + [
              hSigAssignR read [
                hLiteral in_port_1 NOLIST
              ]
              hSigAssignR read [
                hLiteral in_port_2 NOLIST
              ]
            ]
          ]
        ]
      ]
    ]
  ]
